### DISCUSSION TITLE
How to cut an image into equal pieces + join equal pieces

### DISCUSSION THREAD

**Discussion by revyourengines** [ImageMagick/ImageMagick#2820] [Help]
What I want to do is cut an image into 4 equal pieces.
I need the code to measure the size and cut into 4 equal quadrants.

Then I want code to join 4 images (these will be different size than the above).
The images would need to be named accordingly, like 1.png, 2.png, 3.png and 4.png (otherwise the code just joins them in the order given).

Also want all of the above done with 9 images.

Not sure where to start.
Hoping someone can help.
I want to do the above in DOS.

Not sure where to start.
Thanks.

**Comment by t246246** [comment]
Your question is little ambiguous. For example, do you want to divide 2x2 or 3x3 pieces, or 1x4 or 4x1?  What should you do if the image pixel size is not a multiple of 2 or 3?

I suggest you start from the basics of -crop and -append (and +append) command and tell what you have done so far.
Then ask more specific questions or  see complex example as in  https://www.imagemagick.org/Usage/crop/#crop_tile

**Comment by fmw42** [comment]
What version of Imagemagick? Post and example input and output, so we better understand what you want to do.

**Comment by snibgo** [comment]
Perhaps "-crop NxM@" would be useful. For example, to crop into 3 pieces horizontally and 2 vertically (so 6 in all):
```
magick in.png -crop 3x2@ out.png
```
The outputs are named out-0.png etc.

They can be recombined simply:
```
magick out*.png final.png
```
